### PR TITLE
Update JITM Logic

### PR DESCRIPTION
**Pull request by samhotchkiss**
#### Changes proposed in this Pull Request:
Change the logic so that dismissing one JITM only dismisses that one, and not all JITMs

Allow a JITM to come back after 6 months, but if it's dismissed a second time, then banish is forever.

#### Testing instructions:

* Install this branch on your site, deactivate Akismet and VaultPress
* Check to see if a JITM shows on the main comment page.  Dismiss it.
* Publish a post, you should see a JITM for VaultPress.  Dismiss it.
* Go back to the main comment page, that JITM should not display

### CODE REVIEW

**Comment by beaulebens** [comment]
This seems to do something weird/inconsistent. I see the AJAX request go through to dismiss a JITM, and it returns a 200/success. That would suggest that it's saved the fact that I dismissed this notice.

The notice on the Comments page stays away once it's been dismissed.

The notice after publishing a post comes back though, no matter how many times I dismiss it. (publish, dismiss, add new, publish, see notice again).

There's one under Dashboard > Updates which keeps coming back as well. No matter how often I dismiss it, if I refresh the page (or navigate away/back again), it's there as if I never dismissed.

I also noticed that for some reason, we're POSTing a bunch of data when attempting to dismiss a notice. Seems unnecessary and unintentional:

```
action:jitm_ajax
jitmNonce:5888888c99
photon[success]:Success! Photon is now actively optimizing and serving your images for free.
photon[fail]:We are sorry but unfortunately Photon did not activate.
manage[success]:Success! WordPress.com tools are now active.
manage[fail]:We are sorry but unfortunately Manage did not activate.
stats[success]:Success! Stats are now active.
stats[fail]:We are sorry but unfortunately Stats did not activate.
jitm_stats_url:https://pixel.wp.com/g.gif?v=wpcom2&rand=6e28b652&x_jetpack-jitm=wordpresstools
jitmActionToTake:dismiss
jitmModule:vaultpress
```

**Comment by samhotchkiss** [comment]
the data tag was on the wrong element... fixed it :)

File: class.jetpack-jitm.php [hunk 1]
```diff
@@ -61,22 +61,47 @@ function prepare_jitms( $screen ) {
...
+		if ( 'update-core' == $screen->base && ! Jetpack::is_plugin_active( 'vaultpress/vaultpress.php' ) ) {
+			return $this->enqueue_jitm( 'backups_updates' );
 		}
 	}
+	
+	/*
+	 * Checks to see if JITM has been dismissed, and, if not, enqueues it
+	 *
+	 */
+	function enqueue_jitm( $jitm_slug )
```
Comment (dereksmart) [inline class.jetpack-jitm.php]: Opening brace should be on same line

File: class.jetpack-jitm.php [hunk 2]
```diff
@@ -61,22 +61,47 @@ function prepare_jitms( $screen ) {
...
+	{
+		if( ! $jitm_slug ) {
```
Comment (dereksmart) [inline class.jetpack-jitm.php]: Space between `if` and opening parenth

File: class.jetpack-jitm.php [hunk 3]
```diff
@@ -61,22 +61,47 @@ function prepare_jitms( $screen ) {
...
+			return;
+		}
+		
+		$hide_jitm = Jetpack_Options::get_option( 'hide_jitm' );
+		// Don't show the JITM if it's been seen in the last 6 months or dismissed twice in the past.
+		if( is_array( $hide_jitm ) &&
```
Comment (dereksmart) [inline class.jetpack-jitm.php]: Space between `if` and opening parenth

File: class.jetpack.php [hunk 4]
```diff
@@ -603,14 +603,22 @@ function jetpack_jitm_ajax_callback() {
...
+			$jitm_name = $_REQUEST['jitmName'];
 
-			if( ! $jetpack_hide_jitm ) {
+			if( ! $jetpack_hide_jitm || ! is_array( $jetpack_hide_jitm ) ) {
 				$jetpack_hide_jitm = array(
-					$module_slug => 'hide'
+					$jitm_name => array (
+						'last_dismissed' => time(),
+						'dismiss_count'  => 1
+				) );
```
Comment (dereksmart) [inline class.jetpack.php]: These closing parenths should be on separate lines, first one indented one more tab.

File: class.jetpack.php [hunk 5]
```diff
@@ -603,14 +603,22 @@ function jetpack_jitm_ajax_callback() {
...
+			} elseif ( ! isset( $jetpack_hide_jitm[ $jitm_name ] ) || ! is_array( $jetpack_hide_jitm[ $jitm_name ] ) ) {
+				$jetpack_hide_jitm[ $jitm_name ] = array (
+						'last_dismissed' => time(),
```
Comment (dereksmart) [inline class.jetpack.php]: extra tab on this and the line below

**Comment by dereksmart** [comment]
Needs a rebase

**Comment by dereksmart** [comment]
We'll need to change the logic for the newer JITMs like WOO and VideoPress as well

**Comment by dereksmart** [comment]
When dismissing a JITM, I'm seeing this in `wp-admin/error_log` 

```[06-Feb-2017 19:32:41 UTC] PHP Notice:  Undefined variable: module_slug in /home/dereksma/public_html/wp-content/plugins/jetpack/class.jetpack.php on line 627```

File: readme.txt [hunk 6]
```diff
@@ -75,6 +75,7 @@ There are opportunities for developers at all levels to contribute. [Learn more
 
 == Changelog ==
 
+
```
Comment (eliorivero) [inline readme.txt]: Branch currently has conflicts due to this line break so it should be removed since it adds nothing relevant to the code.

**Comment by ebinnion** [comment]
Can we close this in favor of #7058 @withinboredom?

**Comment by withinboredom** [comment]
I need to merge it into #7058

**Comment by withinboredom** [comment]
Implemented in #7058, closing this out

**Pull request closed without merging**